src/ck3/data/provinces.rs

```diff
@@ -128,12 +128,19 @@ impl Ck3Provinces {
         }
     }
 
+    pub(crate) fn verify_exists_provid(&self, provid: ProvId, item: &Token, max_sev: Severity) {
+        if !self.provinces.contains_key(&provid) {
+            let msg = format!("province {provid} not defined in map_data/definition.csv");
+            report(ErrorKey::MissingItem, Item::Province.severity().at_most(max_sev))
+                .msg(msg)
+                .loc(item)
+                .push();
+        }
```

Comment:
The usual way to do this is to call something like `self.provinces.verify_exists(provid)` (possibly passing max_sev in too) so that `self.provinces` can generate the message about it not existing. That way the knowledge about `map_data/definition.csv` can stay encapsulated there.
(Especially since the logic of what makes a provid valid in the game is not 100% known yet)

The interface between `Everything` and the `FileHandler`s works that way too.

Reply:
Yeah the reason why I didn't use that is that normally verify_exists requires a token as the argument but here we are using `ProvId` so I don't want to be confusing. Since we have already parsed the ProvId already, there isn't any point to still pass the token and parse all over again, especially we have verified that it won't generate an error.